### ISSUE TITLE
Magento 2.1.0 RC1 : Undefined index: base_grand_total

### ISSUE THREAD

**Issue by quienti** [magento/magento2#4632]
## Steps to reproduce
1. Update 2.0.6 to 2.1.0 RC1
2. Go to Admin -> Sales -> Invoices
## Expected result
1. Show invoices list
## Actual result
1. Error at loading
   ![sans titre 4](https://cloud.githubusercontent.com/assets/7326954/15465680/28d967ee-20d6-11e6-9bec-c8633a86dd9c.jpg)

**Comment by Vinai** [comment]
Not a lot of additional insight, but might be related:

* Version 2.1.0-rc3
* Installed via `composer create-project` as version 2.0.7 and then upgraded to 2.1.0-rc3
* Visiting the page in the admin works
* But when running the integration test suite, the table column `sales_invoice.base_grand_total` doesn't exist.
* The error occurs often, for example with tests using the fixture `Magento/Sales/_files/invoice.php`

Strangely, if I inspect the table after the test failure, the column `base_grand_total` is present.

Running the same test in an 2.1.0-rc3 installation from a github repository clone succeeds and the error does not happen.

I'm sorry I don't have any more substantial info. Just thought this might help debugging the issue.

**Comment by Vinai** [comment]
The same problem happened to me after upgrading from a 2.0.7 composer project to 2.1.0-rc3 while running the integration tests.

Turned out the DB dump in the sandbox directory was still from 2.0.7, and running the installation over the dump didn't add the column to the table `sales_invoice_grid`.

Once I removed the dump and the table, it got recreated with the `base_grand_total` column.

I didn't dig any deeper, but it seems that under some conditions during an upgrade over an existing table, the script `\Magento\Sales\Setup\InstallSchema` doesn't add the column.
